applications/skyepass.md

```diff
@@ -0,0 +1,251 @@
+
+Users who signup will first create a blockchain wallet and have the mnemonic (and a master password) as their sole identity credentials (pretty standard blockchain wallet stuff). Later, each database instance is called a `vault` (standard name for all password managers) and they are light-weight file based databases ([lowDB](https://github.com/typicode/lowdb) seems to be a great choice). User can be given options to choose the encryption behavior of their database. By default, the vault will be split into some pieces with a Shamir's secret sharing mechanism. 
+
+For instance, for a simplest sharing schema, when the vault is created to be shared with 2 other family members, the vault will be split into 4 parts (we call them `horcrux`, for those who do not know [the Harry Potter reference](http://harrypotter.shoutwiki.com/wiki/Horcrux#:~:text=A%20Horcrux%20is%20a%20powerful,one%20is%20to%20true%20immortality.) ) with a minimum quorum of 2 to be decrypted. One piece will be sent to IPFS without encryption, the other 3 pieces will be encrypted by each member's public key and be sent to IPFS. An NFT will be minted for the owner. The ID of the NFT will be the `vault ID` and the NFT's URI will be a metadata piece that only the owner can change as exampled below: 
+
+```json
+{
+    "pieces": 4,
+    "quorum": 2,
+    "nounce": 18,
+    "owner": "5Ef9ES1SLQZU4KucGsjvs8qexvppQFmDgHiqoqVptJ9nZDeu",
+    "members": [
+        "5EKj6S1SLQZU4KucGsjvs8qexvppQFmDgHiqsdsdtJ9nZ123",
+        "5EJK1S1SLQZjkLKucGsjvs8qekdjpQFmDgHiqoqVptJ9nZ978"
+    ],
+    "unencrypted_cid": "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+    "encrypted_cids": [
+        {
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5EKj6S1SLQZU4KucGsjvs8qexvppQFmDgHiqsdsdtJ9nZ123"
+        },{
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5EJK1S1SLQZjkLKucGsjvs8qekdjpQFmDgHiqoqVptJ9nZ978"
+        }, {
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5Ef9ES1SLQZU4KucGsjvs8qexvppQFmDgHiqoqVptJ9nZDeu"
+        }
+    ]
+}
+```
+
+
+
+The reason why we design such mechanism serves 3 purposes. 
+
+1. Reserve the capacity for advanced users to create more complicated sharing schema. 
+    - For instance, a user can create a vault and assign trustee to take over one's estate when the user passes away. The user can split the vault to 5 `horcrux` and set the minimum decryption quorum to 3. 2 pieces encrypted with the user's own public key, 1 piece encrypted with a trustee A's public key, 1 piece encrypted with another trustee B's public key and 1 last piece to the user's lawyer. In event of death, A and B can go to the lawyer and decrypt the vault and inherit the user's digital identities. 
+    - A team can create a vault that requires 2 members to decrypt a vault, or require the owner's piece to decrypt a vault etc.
```

Comment:
What would that look like, do I need to have both private keys on one device?

Reply:
Nope. I imagine: when a user open the vault that requires two members to unlock, the screen will say `waiting for another user to unlock` and freeze there until another person unlocks the vault at the same time. I'm planning on using some of the [Signal](https://signal.org/docs/) tech stack to communicate between the two clients. 

As I mentioned to Marcin on my earlier replies, we have not put our focus on this part yet. We want to think and investigate it more when we are creating a solution for a specific team. It might involves using a relay server to do so, and we can either host our own servers(expensive, huge energy sucker and huge security scrutiny), or help the team in need to do so on their own private servers as an internal service.